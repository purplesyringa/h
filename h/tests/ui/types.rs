trait WithConst {
    const MEOW: i32 = 1234;
}

impl WithConst for () {}

struct Meow<T>(core::marker::PhantomData<T>);

impl<T> WithConst for Meow<T> {}

fn main() {
    let c_str = h::map! {
        for _;
        c"meow" => (),
    };

    let index = h::map! {
        for _;
        [1][0] => (),
    };

    let integer_literal = h::map! {
        for _;
        0_integer => (),
    };

    let macros = h::map! {
        for _;
        h::map! {
            for _;
            () => (),
        } => (),
    };

    let associated_const = h::map! {
        i32::MAX => (),
        <()>::MEOW => (),
        <() as WithConst>::MEOW => (),
        Meow::<i32>::WithConst => (),
    };
}
