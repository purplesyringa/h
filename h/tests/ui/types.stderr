error: C-string literals are not supported
       `h` is not a full-blown interpreter
  --> tests/ui/types.rs:14:9
   |
14 |         c"meow" => (),
   |         ^^^^^^^

error: cannot perform this indexing operation
       `h` is not a full-blown interpreter; only unsizing `array[..]` slicing is supported
  --> tests/ui/types.rs:19:9
   |
19 |         [1][0] => (),
   |         ^^^^^^

error: invalid integer literal suffix `integer`
  --> tests/ui/types.rs:24:9
   |
24 |         0_integer => (),
   |         ^^^^^^^^^

error: macros are not allowed in key position
  --> tests/ui/types.rs:29:9
   |
29 | /         h::map! {
30 | |             for _;
31 | |             () => (),
32 | |         } => (),
   | |_________^

error: trait paths are not allowed in key position
       `h` is not a full-blown interpreter; only certain constants are supported
  --> tests/ui/types.rs:37:9
   |
37 |         <()>::MEOW => (),
   |         ^^^^^^^^^^

error: trait paths are not allowed in key position
       `h` is not a full-blown interpreter; only certain constants are supported
  --> tests/ui/types.rs:38:9
   |
38 |         <() as WithConst>::MEOW => (),
   |         ^^^^^^^^^^^^^^^^^^^^^^^

error: generic parameters are not allowed in paths in key position
       `h` is not a full-blown interpreter; only certain constants are supported
  --> tests/ui/types.rs:39:9
   |
39 |         Meow::<i32>::WithConst => (),
   |         ^^^^^^^^^^^^^^^^^^^^^^
